vscode_extension/src/providers/AacDefinitionEditorProvider.ts

```diff
@@ -0,0 +1,235 @@
+
+    public saveCustomDocument(document: AacDefinitionsDocument, cancellation: vscode.CancellationToken): Thenable<void> {
+        return document.save(cancellation);
+    }
+
+    public saveCustomDocumentAs(document: AacDefinitionsDocument, destination: vscode.Uri, cancellation: vscode.CancellationToken): Thenable<void> {
+        return document.saveAs(destination, cancellation);
+    }
+
+    public revertCustomDocument(document: AacDefinitionsDocument, cancellation: vscode.CancellationToken): Thenable<void> {
+        return document.revert(cancellation);
+    }
+
+    public backupCustomDocument(document: AacDefinitionsDocument, context: vscode.CustomDocumentBackupContext, cancellation: vscode.CancellationToken): Thenable<vscode.CustomDocumentBackup> {
+        return document.backup(context.destination, cancellation);
+    }
+
+    //#endregion
+
+    /**
+     * Get the editor webview HTML.
+     */
+    private getHtmlForWebview(webview: vscode.Webview): string {
+        // Local path to script and css for the webview
+        const editorSourceDirPath = path.join(this._context.extensionUri.fsPath, 'src', 'editor', 'definitions');
+
+        const scriptUri = webview.asWebviewUri(vscode.Uri.parse(path.join(editorSourceDirPath, 'definitionVisualEditor.js')));
+        const styleMainUri = webview.asWebviewUri(vscode.Uri.parse(path.join(editorSourceDirPath, 'definitionEditor.css')));
+
+        const jsonNodeModulesPath = path.join(this._context.extensionUri.fsPath, 'node_modules');
+
+        const jsonEditorPath = vscode.Uri.parse(path.join(jsonNodeModulesPath, '@json-editor', 'json-editor', 'dist', 'jsoneditor.js'));
+
+        const jsonEditorScriptUri = webview.asWebviewUri(jsonEditorPath);
+
+
+        // Use a nonce to whitelist which scripts can be run
+        const nonce = getNonce();
+
+        return `
+			<!DOCTYPE html>
+			<html lang="en">
+			<head>
+				<meta charset="UTF-8">
+
+				<!--
+				Use a content security policy to only allow loading images from https or from our extension directory,
+				and only allow scripts that have a specific nonce.
+				-->
+				<meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src ${webview.cspSource} blob:; style-src ${webview.cspSource} 'unsafe-inline'; script-src * data: ${webview.cspSource} 'unsafe-inline' 'unsafe-eval';">
+
+				<meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+				<link href="${styleMainUri}" rel="stylesheet" />
+
+                <script src="${jsonEditorScriptUri}" http-equiv="Content-Security-Policy" script-src-elem "${jsonEditorScriptUri}" nonce="${nonce}";></script>
+				<title></title>
+			</head>
+			<body>
+				<div id="main"></div>
+                <button id='submit'>Save</button>
+                <script src="${scriptUri}" http-equiv="Content-Security-Policy" script-src-elem 'unsafe-inline' ${scriptUri} nonce='${nonce}';></script>
+			</body>
+			</html>`;
```

Comment:
The formatting looks off here - is there a reason to use tabs, instead of spaces?

Reply:
Nah, good catch.